scripts/ci-entrypoint.sh

```diff
@@ -150,10 +145,24 @@ install_addons() {
     | sed 's/namespace: kube-system/namespace: calico-system/' \
     | ${KUBECTL} create -f -
 
-    # install cni
+    # install Calico CNI
     echo "Installing Calico CNI via helm"
+    if [[ "${CIDR0}" =~ .*:.* ]]; then
+        echo "Cluster CIDR is IPv6"
+        CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico-ipv6/values.yaml"
+        CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0}"
+    elif [[ "${CIDR1}" =~ .*:.* ]]; then
+        echo "Cluster CIDR is dual-stack"
+        CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico-dual-stack/values.yaml"
+        CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0},installation.calicoNetwork.ipPools[1].cidr=${CIDR1}"
+    else
+        echo "Cluster CIDR is IPv4"
+        CALICO_VALUES_FILE="${REPO_ROOT}/templates/addons/calico/values.yaml"
+        CIDR_STRING_VALUES="installation.calicoNetwork.ipPools[0].cidr=${CIDR0}"
+    fi
+
     "${HELM}" repo add projectcalico https://projectcalico.docs.tigera.io/charts
-    "${HELM}" install calico projectcalico/tigera-operator -f "${CALICO_VALUES:-${REPO_ROOT}/templates/addons/calico/values.yaml}" --namespace tigera-operator --create-namespace
+    "${HELM}" install calico projectcalico/tigera-operator -f "${CALICO_VALUES_FILE}" --set-string "${CIDR_STRING_VALUES}" --namespace tigera-operator --create-namespace
```

Comment:
Looks like this `installation.calicoNetwork.ipPools[0].cidr` config override is totally new w/ this PR? Does that suggest that even in the ipv4 scenario we've been doing this wrong previously?

Reply:
not entirely. Before it was using the value from https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/addons/calico/values.yaml#L8 which happens to be the default ipv4 CIDR we use in all test templates

Reply:
So without this change, is IPv6 test doing wrong?

Reply:
not in release v1.6 and below (it would be wrong if we release v1.7 without this change though). The changes to install Calico with Helm are only in the main branch currently.

Reply:
Got it. Thanks.